zeph-core tests fail without explicit --features env-vault after #3479

## Description
After PR #3479 gated `EnvVaultProvider` behind the `env-vault` feature, running `cargo nextest run -p zeph-core` (without explicit feature flags) fails with a compilation error.

This regresses the developer workflow: running tests for a single crate without specifying feature flags is a common pattern documented in CLAUDE.md.

## Reproduction Steps
1. `cargo nextest run -p zeph-core` (no `--features` flag)

Observe:
```
error[E0432]: unresolved import `zeph_vault::EnvVaultProvider`
   --> crates/zeph-core/src/lib.rs:153:13
    |
153 |     pub use zeph_vault::EnvVaultProvider;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `EnvVaultProvider` in the root
```

## Expected Behavior
`cargo nextest run -p zeph-core` should compile and run tests without explicit feature flags.

## Actual Behavior
Compilation fails because `crates/zeph-core/src/lib.rs:153` is gated with `#[cfg(any(test, feature = "env-vault"))]`, which activates during test builds (`cfg(test)` is true), but the `zeph-vault/env-vault` feature is not enabled unless explicitly passed.

## Root Cause
`#[cfg(any(test, feature = "env-vault"))]` in `zeph-core/src/lib.rs` uses the `test` cfg to include the `pub use zeph_vault::EnvVaultProvider;` re-export in test builds. However, `zeph-vault` only exports `EnvVaultProvider` when `zeph-vault/env-vault` feature is active. Since `zeph-core`'s `env-vault` feature (`= ["zeph-vault/env-vault"]`) is not in `zeph-core`'s `[features] default`, running tests without explicit `--features env-vault` leaves `zeph-vault::EnvVaultProvider` invisible.

## Fix Options
1. **Preferred**: Change `#[cfg(any(test, feature = "env-vault"))]` to `#[cfg(feature = "env-vault")]` — require explicit feature for test use too. Tests that need `EnvVaultProvider` must run with `--features env-vault`.
2. **Alternative**: Add `env-vault` to `zeph-core`'s default features list.

## Environment
- Version: 0.20.0 (HEAD 55044a20)
- Introduced by: PR #3479
- Workaround: `cargo nextest run --config-file .github/nextest.toml -p zeph-core --features env-vault`

## Logs / Evidence
```
error[E0432]: unresolved import `zeph_vault::EnvVaultProvider`
   --> crates/zeph-core/src/lib.rs:153:13
153 |     pub use zeph_vault::EnvVaultProvider;
hint: a similar name exists in the module - pub use zeph_vault::VaultProvider;
```
Full build passes: `cargo nextest run --workspace --features full` → 8962/8962 tests PASS.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 ### Fixed
 
+- fix(core): remove `test` arm from `EnvVaultProvider` re-export cfg gate in `zeph-core/src/lib.rs`
+  so that `cargo nextest run -p zeph-core` (no explicit `--features env-vault`) compiles cleanly
+  (#3485, regressed by #3479).
+- fix(bootstrap): `build_single_provider_from_pool` now skips embed-only entries when selecting
+  the fallback provider index, preventing an embed model from being used for chat when no provider
+  has `default = true` (#3484).
+
 - fix(focus): `complete_focus` called in a batch with other tools no longer orphans the
   current-turn tool results (#3476). The truncation in `complete_focus_tool` now preserves
   the current turn's assistant `tool_calls` message so that the subsequent `User(tool_results)`

--- a/crates/zeph-core/src/lib.rs
+++ b/crates/zeph-core/src/lib.rs
@@ -149,7 +149,7 @@ pub mod vault {
     /// This provider reads secrets from process environment variables and is
     /// intended **exclusively for development and testing**. Never enable this
     /// feature in production builds. Use [`AgeVaultProvider`] instead.
-    #[cfg(any(test, feature = "env-vault"))]
+    #[cfg(feature = "env-vault")]
     pub use zeph_vault::EnvVaultProvider;
 
     #[cfg(any(test, feature = "mock"))]

--- a/src/bootstrap/provider.rs
+++ b/src/bootstrap/provider.rs
@@ -520,7 +520,11 @@ fn build_single_provider_from_pool(
     pool: &[ProviderEntry],
     config: &Config,
 ) -> Result<AnyProvider, BootstrapError> {
-    let primary_idx = pool.iter().position(|e| e.default).unwrap_or(0);
+    let primary_idx = pool
+        .iter()
+        .position(|e| e.default)
+        .or_else(|| pool.iter().position(|e| !e.embed))
+        .unwrap_or(0);
     let primary = &pool[primary_idx];
     match build_provider_from_entry(primary, config) {
         Ok(p) => Ok(p),

